examples/injection/module/rust/src/lib.rs

```diff
@@ -0,0 +1,229 @@
+};
+use proto::{
+    blob_request::Request, BlobRequest, BlobResponse, BlobStore, BlobStoreDispatcher,
+    BlobStoreInterface, BlobStoreProviderSender, BlobStoreRequest, BlobStoreSender, GetBlobRequest,
+    PutBlobRequest,
+};
+
+oak::entrypoint!(grpc_fe => |_in_channel| {
+    oak::logger::init_default();
+    let (to_provider_write_handle, to_provider_read_handle) = oak::channel_create().unwrap();
+    let (from_provider_write_handle, from_provider_read_handle) = oak::channel_create().unwrap();
+    oak::node_create(&oak::node_config::wasm("app", "provider"), to_provider_read_handle)
+        .expect("Failed to create provider");
+    Sender::new(to_provider_write_handle)
+        .send(&BlobStoreProviderSender { sender: Some(Sender::new(from_provider_write_handle)) })
+        .expect("Failed to send handle to provider");
+
+    let frontend = BlobStoreFrontend::new(
+            Sender::new(to_provider_write_handle),
+            Receiver::new(from_provider_read_handle));
+    let dispatcher = BlobStoreDispatcher::new(frontend);
+    let grpc_channel = oak::grpc::server::init("[::]:8080")
+        .expect("could not create gRPC server pseudo-Node");
+    oak::run_event_loop(dispatcher, grpc_channel);
+});
+
+oak::entrypoint!(provider => |frontend_read| {
+    oak::logger::init_default();
+    let frontend_sender =
+        Receiver::<BlobStoreProviderSender>::new(frontend_read).receive()
+            .expect("Did not receive a write handle")
+            .sender
+            .expect("No write handle in received message");
+    oak::run_event_loop(BlobStoreProvider::new(frontend_sender), frontend_read);
+});
+
+oak::entrypoint!(store => |reader| {
+    oak::logger::init_default();
+    let sender =
+        Receiver::<BlobStoreSender>::new(reader).receive()
+            .expect("Did not receive a write handle")
+            .sender
+            .expect("No write handle in received message");
+    oak::run_event_loop(BlobStoreImpl::new(sender), reader);
+});
+
+enum CachedStore {
+    NotCached {
+        sender: Sender<BlobStoreRequest>,
+        receiver: Receiver<BlobStoreInterface>,
+    },
+    Cached(BlobStoreInterface),
+}
+
+struct BlobStoreFrontend {
+    store: CachedStore,
+}
+
+impl BlobStoreFrontend {
+    pub fn new(
+        sender: Sender<BlobStoreRequest>,
+        receiver: Receiver<BlobStoreInterface>,
+    ) -> BlobStoreFrontend {
+        BlobStoreFrontend {
+            store: CachedStore::NotCached { sender, receiver },
+        }
+    }
+
+    fn get_interface(&mut self) -> &BlobStoreInterface {
+        // Make sure it is cached
+        if let CachedStore::NotCached { sender, receiver } = &self.store {
+            sender
+                .send(&BlobStoreRequest {})
+                .expect("Failed to send BlobStoreRequest");
+            let iface = receiver
+                .receive()
+                .expect("Failed to receive BlobStoreInterface");
+            self.store = CachedStore::Cached(iface);
```

Comment:
Be nice to do `sender.close()` and `receiver.close()` as the handles are about to be lost anyway.